packages/lesswrong/lib/collections/users/helpers.ts

```diff
@@ -37,7 +37,8 @@ export const userOwnsAndInGroup = (group: string) => {
 
 export const userIsSharedOn = (currentUser: DbUser|UsersMinimumInfo|null, document: PostsList|DbPost): boolean => {
   if (!currentUser) return false;
-  return document.shareWithUsers && document.shareWithUsers.includes(currentUser._id)
+  return document.shareWithUsers?.includes(currentUser._id) ||
+    document.coauthorStatuses?.findIndex(({ userId }) => userId === currentUser._id) >= 0;
```

Comment:
This assumes that the database migration has been run. In the absence of the migration it's possible to add a small hack here for backwards compatibility, but the migration is obviously preferred.